editor/plugins/gdextension/cpp_scons/cpp_scons_gdext_creator.cpp

```diff
@@ -0,0 +1,232 @@
+	_write_file(_res_path.path_join("doc_classes/ExampleNode.xml"), EXAMPLENODE_XML);
+	_write_file(_res_path.path_join("icons/ExampleNode.svg"), EXAMPLENODE_SVG);
+	_write_file(_res_path.path_join("icons/ExampleNode.svg.import"), EXAMPLENODE_SVG_IMPORT);
+	_write_file(_res_path.path_join("src/.gdignore"), "");
+	_write_file(_res_path.path_join(".gitignore"), GDEXT_GITIGNORE + "*.obj");
+	_write_file(_res_path.path_join(_library_name + ".gdextension"), LIBRARY_NAME_GDEXTENSION);
+}
+
+void CppSconsGDExtensionCreator::_write_gdext_only_files() {
+	_ensure_file_contains("res://.gitignore", "*.dblite");
+	_write_file(_res_path.path_join("src/example_node.cpp"), EXAMPLE_NODE_CPP);
+	_write_file(_res_path.path_join("src/example_node.h"), EXAMPLE_NODE_H);
+	_write_file(_res_path.path_join("src/register_types.cpp"), REGISTER_TYPES_CPP);
+	_write_file(_res_path.path_join("src/register_types.h"), REGISTER_TYPES_H);
+	_write_file(_res_path.path_join("src/" + _library_name + "_defines.h"), GDEXT_DEFINES_H);
+	_write_file(_res_path.path_join("src/initialize_gdextension.cpp"), INITIALIZE_GDEXTENSION_CPP.replace("#include \"__UPDIR_DOTS__/../", "#include \""));
+	_write_file(_res_path.path_join("SConstruct"), SCONSTRUCT_ADDON.replace(" + Glob(\"__UPDIR_DOTS__/*.cpp\")", "").replace(",__UPDIR_DOTS__/", ""));
+}
+
+void CppSconsGDExtensionCreator::_write_gdext_module_files() {
+	_ensure_file_contains("res://.gitignore", GDEXT_GITIGNORE);
+	_make_dir_in_res("res://tests");
+	_write_file("res://SCsub", SCSUB);
+	_write_file("res://config.py", CONFIG_PY);
+	_write_file("res://example_node.cpp", EXAMPLE_NODE_CPP);
+	_write_file("res://example_node.h", EXAMPLE_NODE_H);
+	_write_file("res://register_types.cpp", REGISTER_TYPES_CPP);
+	_write_file("res://register_types.h", REGISTER_TYPES_H);
+	_write_file("res://" + _library_name + "_defines.h", SHARED_DEFINES_H);
+	_write_file("res://tests/test_" + _base_name + ".h", TEST_BASE_NAME_H);
+	_write_file("res://tests/test_example_node.h", TEST_EXAMPLE_NODE_H);
+	_write_file(_res_path.path_join("src/initialize_gdextension.cpp"), INITIALIZE_GDEXTENSION_CPP);
+	_write_file(_res_path.path_join("SConstruct"), SCONSTRUCT_ADDON);
+}
+
+void CppSconsGDExtensionCreator::create_gdextension(const String &p_path, const String &p_base_name, const String &p_library_name, int p_variation_index, bool p_compile) {
+	_res_path = p_path;
+	_base_name = p_base_name;
+	_library_name = p_library_name;
+	_updir_dots = String("../").repeat(p_path.count("/", 6)) + "..";
+	_strip_module_defines = p_variation_index == 0;
+	_write_common_files_and_dirs();
+	if (p_variation_index == 0) {
+		_write_gdext_only_files();
+	} else {
+		_write_gdext_module_files();
+	}
+	_git_clone_godot_cpp(p_path.path_join("src"), p_compile);
+}
+
+PackedStringArray CppSconsGDExtensionCreator::get_language_variations() const {
+	PackedStringArray variants;
+	variants.push_back("C++ with SCons, GDExtension only"); // 0
```

Comment:
You could add an enum for the variations instead of relying on numbers.

Reply:
Ok, added enum `CppSconsGDExtensionCreator::LanguageVariation`.